### ISSUE TITLE
Issue in Thumbnail synchronization

### ISSUE THREAD

**Issue by bane92in** [fastaddons/GroupSpeedDial#274]
**Describe the bug**
I use custom thumbnails in my browser in my primary PC, which are not getting synced to the cloud. The dials are getting synced flawlessly though. It's just the thumbnails that refuse to get updated. As a result the thumbnails of the dials in my secondary PC are also out of sync i.e., being generated from the webpage.
I can confirm that the sync issues are in my cloud dials, as the newly updated thumbnails in the cloud are just blurry logos (fetched off the internet?). Interestingly though my older dials are perfectly synced to the cloud. It's just the recently updated ones are having the issue

**Device info:**
Primary PC
 - operating system - Win 11
 - Browser and version - Firefox v116.0.2 (64 bit)
 - GroupSpeedDial version 23.0: 

Secondary PC specs are the same.

**Additional info**
Add any other info about the problem here.

**Comment by Juraj-Masiar** [comment]
Hello,
Thank you for the report! I'll look into it because I think you are not alone.
For now, to force a single thumbnail to the cloud, you can do - right click / Advanced / Restore previous dial. And then once again to revert it. That will count as "update" of the thumbnail so it will be later synchronized to the cloud.
But easier would be probably to export dials and thumbnails to a file (from Options page / Import backup tab) and then import it on the other device.

But before you do that, could you answer these questions:
1. right click one of the affected dials and select "Edit dial" and go to Auto-reload tab - is the "Auto-reload thumbnail" enabled? It should NOT be enabled for dials with custom thumbnail.
2. how did you set the custom thumbnail? Also check the Edit dial / Thumbnail tab and see which option is selected.
3. was this working before?
4. did you used import / export from file before? Or how did you set it up on the secondary PC?

**Comment by bane92in** [comment]
Thanks for the quick response.

Tried the method of forcing single thumbnail to the cloud as you suggested. Waited for half an hour. Didn't work.

Export/Import method should work as it bypasses the cloud. Haven't tried it yet though.

Query responses:
1. Auto Reload Thumbnail is Disabled for all the affected dials.
2. I had setup the custom thumbnails in my primary PC vide Edit Dial> Thumbnail> Custom Thumbnail (This option is still selected)
3. YES it was working before. That's why i find it surprising. 
4. I haven't used the Export/Import feature yet.
Couple of years ago, I had set up most of the dials and custom thumbnails initially in my primary PC. Next I made a Cloud account and enabled sync. Few days later I installed your addon to my Secondary PC and it synced with the cloud without any hiccup (Dials and thumbnails both)

A couple of days back I started editing the existing dial URLs, the dial positions and thumbnails in my secondary PC and ensured sync is enabled. Later in my primary PC the URLs and dial positions got updated but the thumbnails wouldn't.  

On checking the Cloud dials i am seeing the URL's and dial positions are as it should be in sync. But in the edited dials the thumbnails are either missing or replaced with a blurry generic logo auto generated form the internet i assume.

I also tried the other way round. Edited some other thumbnails in my Primary PC but it wont sync to the cloud dials. Hence couldn't get it updated to my secondary PC.

Hope this information helps

**Comment by Juraj-Masiar** [comment]
> Tried the method of forcing single thumbnail to the cloud as you suggested. Waited for half an hour. Didn't work.

Yeah, thumbnails are synced only every few hours (to save resources). But you can trigger thumbnails sync by clicking "synced" label in the cloud icon on the dials page.

> A couple of days back I started editing the existing dial URLs, the dial positions and thumbnails in my secondary PC 

**Are these the affected dials?**
If yes then this is a great news because I can track down the issue easily!
Changing URL of the dial is a "big deal" under the hood since thumbnails are linked to the specific URL in the specific group. So when you change URL the thumbnail is actually deleted and recreated and needs to be re-uploaded to the cloud. That would explain why it disappeared from the server. Now just find out why the new one wasn't uploaded.

**Comment by Juraj-Masiar** [comment]
So I've checked the code and also tested it manually and everything seems to be working fine.
Possible causes I can think of - if you change URL on a secondary PC where the thumbnails sync. is not enabled, it will remove the thumbnail from the main PC (since changing URL means it's a "new" dial).
If the thumbnails sync. is enabled on both devices, then maybe the image upload failed somehow - try to add a new dial and then click the "synced" label as I mentioned before to see if it can upload it to the cloud. And if not, then open main menu / Advanced / Send error logs... and check if you can see errors there.
